### PR TITLE
add Mekotronics-R58

### PR DESCRIPTION
**Pull request by rpardini**
DTS's, a hack for `stmmac` LEDs, and that pesky `chosen` `bootargs` root reference that had popped up before but lost in time. DT's and stmmac hack provided by @monkaBlyat.

### CODE REVIEW

**Comment by rpardini** [comment]
Force pushed, updated some Mekotronic product names in the DT.

File: arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi
Comment (amazingfate) [inline arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi:1]: We don't have to change this upstream device tree. We can treat it downstream like rock5b: https://github.com/armbian/linux-rockchip/blob/rockchip-5.10/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts#L29

**Review by amazingfate — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi
Comment (rpardini) [reply arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi:1]: Agreed! Will do.

Comment (rpardini) [reply arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi:1]: Done...

**Review by amazingfate — APPROVED** [review]

**Pull request merged**